styles/01-css-approaches.mdx

```diff
@@ -0,0 +1,111 @@
+## CSS approaches
+
+* CSS is all in the global scope by default
+* CSS-in-JS approaches with React let you scope your CSS to individual components
+* Functional CSS approaches let you avoid writing redundant CSS
+
+---
+
+## CSS-in-JS
+
+* Approaches that move some part of the CSS process into JavaScript
+* Normally try to solve the problem of CSS being in the global scope
+* Can require more set up, but end up more flexible in the long term
+* Read into CSS Modules and Styled Components if this method is of interest
+
+---
+
+## Functional CSS
+
+* AKA Atomic CSS
+* Many small, single-purpose classes
+* Write less CSS and stay more consistent by using what already exists
+* Libraries like Tachyons, Tailwind and Bulma
+
+---
+
+## Tachyons
+
+```css
+.flex {
+    display: flex;
+}
+.flex-row {
+    flex-direction: flex-row;
+}
+.pl4 {
+    padding-left: 4rem;
+}
+.f3 {
+    font-size: 3rem;
+}
+```
+
+```notes
+https://tachyons.io/#style
+```
+---
+
+```javascript
```

Comment:
I like this general idea - but wondering if maybe taking an example that may need a few more classes, and could show a more 'traditional approach'

```js
const Menu = ({ items }) => (
    <ul className="menu-list">
        {items.map((item, i) => (
            <li className="menu-item" key={i}>{item.name}</li>
        ))
    </ul>
```

then could turn into

```js
const Menu = ({ items }) => (
    <ul className="flex flex-row f3">
        {items.map((item, i) => (
            <li className="pl4" key={i}>{item.name}</li>
        ))
    </ul>
```
and - then 'what would be your `menu-list / menu item css classes, become your react components`

```js
const Menu = ({ items }) => (
    <MenuList>
        {items.map((item, i) => (
            <MenuItem key={i}>{item.name}</MenuItem>
        ))
    </MenuList>
```
maybe that's going too fine grained? I just remember the thing that made tachyons click as an 'ah-hah' for me was 'ah, that class-name soup that I was wanting class menu-item for, becomes a component"

Reply:
I will add this!